### PR TITLE
Find text

### PR DESCRIPTION
**Pull request by Maxwell-Yang-PDFTron**
Function:
findText
cancelFindText
getSelection

Prop:
onTextSearchStart
onTextSearchResult

### CODE REVIEW

**Comment by HelloTiago** [comment]
@Maxwell-Yang-PDFTron This findText function is very very close to something I had been looking for for a while! Would it be possible to highlight the text selection(s) that have been found?

Right now if I `console.log` the text selection from `onTextSearchResult` I get the result but there's nothing in the document itself that points me to the selected text.

The native SDK has a method that would be really useful here, [highlightSelection](https://www.pdftron.com/api/ios/Classes/PTPDFViewCtrl.html#/c:objc(cs)PTPDFViewCtrl(im)highlightSelection:withColor:) - is that something in the works?

Thanks!

File: ios/RNTPTDocumentViewManager.m [hunk 1]
```diff
@@ -562,6 +562,30 @@ - (void)behaviorActivated:(RNTPTDocumentView *)sender action:(NSString *)action
     }
 }
 
+- (void)textSearchStart:(RNTPTDocumentView *)sender
+{
+    if (sender.onChange) {
+        sender.onChange(@{
```
Comment (JamesTron) [inline ios/RNTPTDocumentViewManager.m:568]: What in God's name is this? A C function?

File: ios/RNTPTDocumentViewManager.m [hunk 2]
```diff
@@ -809,6 +863,37 @@ - (double)getZoom:(NSNumber *)tag
...
+}
+
+- (void)cancelFindTextForDocumentViewTag:(NSNumber *)tag
+{
+    RNTPTDocumentView *documentView = self.documentViews[tag];
+    if (documentView) {
+        [documentView cancelFindText];
+    } else {
+        @throw [NSException exceptionWithName:NSInvalidArgumentException reason:@"Unable to find DocumentView for tag" userInfo:nil];
+    }
```
Comment (JamesTron) [inline ios/RNTPTDocumentViewManager.m:883]: This pattern shows up a lot. You could probably refactor these into a single method `getDocumentViewForTag` that throws if there isn't one.

**Review by JamesTron — CHANGES_REQUESTED** [review]

**Review by sgong-pdftron — APPROVED** [review]

File: ios/RNTPTDocumentViewManager.m (hunk 2 above)
Comment (Maxwell-Yang-PDFTron) [reply ios/RNTPTDocumentViewManager.m:1075]: I would probably do a refactoring after all PRs have been merged. Thanks.

**Pull request merged**